src/main/java/org/openmrs/addonindex/scheduled/FetchCoreVersionsList.java

```diff
@@ -0,0 +1,80 @@
+package org.openmrs.addonindex.scheduled;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.openmrs.addonindex.domain.artifactory.ArtifactoryFolderInfoChild;
+import org.openmrs.addonindex.domain.artifactory.CoreVersionsSummary;
+import org.openmrs.addonindex.service.artifactory.VersionsService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StreamUtils;
+
+import java.nio.charset.Charset;
+import java.util.stream.Collectors;
+
+@Component
+public class FetchCoreVersionsList {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Value("${core_version_list.url}")
+    private String url;
+
+    @Value("${core_version_list.strategy}")
+    private FetchCoreVersionsList.Strategy strategy = FetchCoreVersionsList.Strategy.FETCH;
+
+    private RestTemplateBuilder restTemplateBuilder;
+
+    private ObjectMapper mapper;
+
+    private VersionsService versionsService;
+
+    @Autowired
+    public FetchCoreVersionsList(RestTemplateBuilder restTemplateBuilder,
+                                 ObjectMapper mapper,
+                                 VersionsService versionsService) {
+        this.restTemplateBuilder = restTemplateBuilder;
+        this.mapper = mapper;
+        this.versionsService = versionsService;
+    }
+
+    @Scheduled(
+            initialDelayString = "${scheduler.fetch_core_versions_list.initial_delay}",
+            fixedDelayString = "${scheduler.fetch_core_versions_list.period}")
+    public void fetchCoreVersionsList() throws Exception {
+        logger.info("Fetching list of OpenMRS-Core versions to index");
+
+        String json;
+        if (strategy == Strategy.LOCAL) {
+            logger.debug("LOCAL strategy");
+            json = StreamUtils.copyToString(getClass().getClassLoader().getResourceAsStream("openmrs-core-versions.json"),
+                    Charset.defaultCharset());
+        } else {
+            logger.debug("FETCH strategy: " + url);
+            json = restTemplateBuilder.build().getForObject(url, String.class);
+        }
+        CoreVersionsSummary children;
+        try {
+            children = mapper.readValue(json, CoreVersionsSummary.class);
+        } catch (Exception ex) {
+            throw new RuntimeException("File downloaded from " + url + " could not be parsed", ex);
```

Comment:
PS- a random google hit about this: https://www.securecoding.cert.org/confluence/display/java/ERR07-J.+Do+not+throw+RuntimeException%2C+Exception%2C+or+Throwable